.github/workflows/ci.yml

```diff
@@ -48,6 +48,11 @@ jobs:
           # queries: ./path/to/local/query, your-org/your-repo/queries@main
           languages: ${{ matrix.language }}
 
+      - if: matrix.language == 'cpp' || matrix.language == 'csharp'
+        name: Build C
+        run: |
+          make   
```

Comment:
> The error 'CodeQL detected code written in Java/Kotlin, Go and c, but not any written in C/C++' is still reported in CI. I saw that the CI result shows that this step has been skipped. Is it because this judgment condition requires the addition of `matrix. language=='c'`?
> 
> CI仍然报这个错"CodeQL detected code written in Java/Kotlin, Go and c, but not any written in C/C++"。我看CI结果显示这一步跳过了，是不是因为这个判断条件需要加上`matrix.language =='c'`？

cpp就是c/c++，CodeQL里面有显示它的语言设置

Reply:
https://github.com/xwm1992/EventMesh/pull/11
这是我自己仓库的pr修改，显示ci check是通过了的，但是提交到主仓库就失败了

Reply:
> 提交到主仓库就失败了

It should be a side effect of these unsynchronized commits: https://github.com/xwm1992/EventMesh/compare/master...apache:eventmesh:master

---

应该是这些没同步的commit带来的副作用：https://github.com/xwm1992/EventMesh/compare/master...apache:eventmesh:master

Reply:
> [xwm1992#11](https://github.com/xwm1992/EventMesh/pull/11) 这是我自己仓库的pr修改，显示ci check是通过了的，但是提交到主仓库就失败了


会不会跟make的路径有关，在根目录下面不能进行make 编译处理C而需要到sdk Makefile文件目录才能
-- 
Could it be related to the path of make? Compilation processing of C cannot be performed under the root directory by make, but requires access to the directory of the sdk Makefile file.

Reply:
In #4323, c sdk adds some third-party libraries using submodules. git doesn't clone or use submodules by default. Needs to make or scan them after adding them.

---

在 #4323 中，c sdk使用了submodule的方式添加了一些第三方库。git默认是不clone或使用submodule的，需要添加它们之后再make或扫描。

Reply:
https://github.com/github/codeql/issues/14703
I raised this issue to the codeql community. We can follow it and reply under this issue.

---
我向codeql 社区提了这个问题的issue，大家可以关注并在这个issue下面回复一下

Reply:
The matrix does not define a `language` property, so `matrix.language` evaluates to `null`.